### ISSUE TITLE
Consumer API: add acknowledge in batch

### ISSUE THREAD

**Issue by ShahOdin** [apache/pulsar#6018]
**Is your feature request related to a problem? Please describe.**

I want to be able to process messages in batches and in parallel. One of the selling features of Pulsar for me, is being able to acknowledge messages separately which allows me to avoid re-processing messages if one message in a batch failed to be processed. However, I don't want to send acknowledgements for each. I'd like to wait on all tasks, see which ones succeeded and then acknowledge those in one commit. 

**Describe the solution you'd like**

Currently the Consumer Api has: `acknowledgeCumulative` which works the same way that Kafka's offsets work. but that doesn't work here. I'm suggesting the follwoing method: 

`
CompletableFuture<Void> acknowledgeInBatchAsync(List<MessageId> messageIds)
`

and the same for nacks. 

**Describe alternatives you've considered**
I think what I want can still be achieved by the following: Let's say I have messages with Ids (1,2,3,4,5) and (2,3,4,5) are processed successfully while (1) isn't. Please confirm if I can achieve what I want with the following: 

```
acknowledgeCumulative(5)
negativeAcknowledge(1)
```

for some reason, I can't find the method `negativeAcknowledge` on the [consumer api](https://pulsar.apache.org/api/client/org/apache/pulsar/client/api/Consumer.html).

**Comment by jiazhai** [comment]
@ShahOdin  regarding the api link, it has some issue and @tuteng is fixing this in Issue #5037 
you could add the release version manually. e.g. https://pulsar.apache.org/api/client/2.4.2/org/apache/pulsar/client/api/Consumer.html

**Comment by jiazhai** [comment]
And you are right, the workaround above is good. 

Regarding this feature,  it seems that without pulsar code change, we could also achieve it by application  side code changes?  e.g.  call several acknowledgeAsync, and wait for all the future complete?

**Comment by ShahOdin** [comment]
> we could also achieve it by application side code changes? e.g. call several acknowledgeAsync, and wait for all the future complete?

Well that was the obvious solution which I ruled out as I wanted to avoid making all those calls.

**Comment by tisonkun** [comment]
Closed as stale. Please create a new issue if it's still relevant to the maintained versions.

IIRC we have cumulative ack already: https://pulsar.apache.org/docs/2.10.x/concepts-messaging/#acknowledgement
